host/xtest/pkcs11_1000.c

```diff
@@ -716,15 +733,110 @@ static CK_RV test_already_initialized_token(ADBG_Case_t *c, CK_SLOT_ID slot)
 	return rv;
 }
 
-static CK_RV test_uninitialized_token(ADBG_Case_t *c, CK_SLOT_ID slot)
+static CK_UTF8CHAR test_acl_auth_token_label[] = "PKCS11 TA test token for ACL based authentication";
+
+static CK_RV init_test_token_acl_auth(CK_SLOT_ID slot)
+{
+	return C_InitToken(slot, NULL, 0, test_acl_auth_token_label);
+}
+
+#define TEE_UUID_NS_NAME_SIZE  128
+#define ACL_PIN_LEN  43
+
+/*
+ * TEE client UUID name space identifier
+ * Client UUIDs are derived from this UUID (which is defined in the Linux
+ * kernel) using the UUIDv5 scheme.
+ */
+static const char *client_uuid_linux_ns = "58ac9ca0-2086-4683-a1b8-ec4bc08e01b6";
+
+static CK_RV init_user_test_token_acl_auth(ADBG_Case_t *c, CK_SLOT_ID slot)
+{
+	CK_FLAGS session_flags = CKF_SERIAL_SESSION | CKF_RW_SESSION;
+	CK_SESSION_HANDLE session = CK_INVALID_HANDLE;
+	CK_RV rv = CKR_GENERAL_ERROR;
+	TEEC_Result result = TEEC_ERROR_GENERIC;
+	TEEC_UUID expected_client_uuid = { };
+	TEEC_UUID uuid_ns = { };
+	char uuid_name[TEE_UUID_NS_NAME_SIZE] = { };
+	char test_token_acl_auth_pin[ACL_PIN_LEN];
+	char *client_id_s = NULL;
+
+	rv = C_OpenSession(slot, session_flags, NULL, 0, &session);
+	if (rv)
+		return rv;
+
+	rv = C_Login(session, CKU_USER, NULL, 0);
+	if (rv == CKR_OK) {
+		C_Logout(session);
+		C_CloseSession(session);
+		return rv;
+	}
+
+	rv = C_Login(session, CKU_SO, NULL, 0);
+	if (rv) {
+		C_CloseSession(session);
+
+		rv = init_test_token_acl_auth(slot);
+		if (rv)
+			return rv;
+
+		rv = C_OpenSession(slot, session_flags, NULL, 0, &session);
+		if (rv)
+			return rv;
+
+		rv = C_Login(session, CKU_SO, NULL, 0);
+		if (rv) {
+			C_CloseSession(session);
+			return rv;
+		}
+	}
+
+	rv = CKR_GENERAL_ERROR;
+
+	result = xtest_uuid_from_str(&uuid_ns, client_uuid_linux_ns);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result)) goto out;
+
+	sprintf(uuid_name, "gid=%x", getegid());
```

Comment:
This looks ok to me seen the 128byte provisionned for `uuid_name` but I wonder whether it would be better to use snprintf()` for and check return value.